staging/src/k8s.io/apiserver/pkg/endpoints/handlers/create.go

```diff
@@ -163,6 +163,17 @@ func createHandler(r rest.NamedCreater, scope *RequestScope, admit admission.Int
 
 		userInfo, _ := request.UserFrom(ctx)
 
+		// ensure namespace on the object is correct, or error if a conflicting namespace was set in the object
+		objectMeta, err := meta.Accessor(obj)
```

Comment:
breadcrumbs for me to chase.  I don't recall us previously enforcing objects to have objectmeta, so https://github.com/openshift/api/blob/master/authorization/v1/types.go#L238-L243 doesn't have it for instance. Perhaps it overrides this entire flow?  i wouldn't expect it to because that would bypass audit, but maybe.

Reply:
fair... if we can't get an accessor, I think it's reasonable to skip this here.

We still have the backstop in staging/src/k8s.io/apiserver/pkg/registry/rest/create.go and staging/src/k8s.io/apiserver/pkg/registry/rest/update.go that require Accessor and repeat the `EnsureObjectNamespaceMatchesRequestNamespace` check before persisting regular objects (types without objectmeta that implement their own Create impl skip *that* flow)